src/main/scala/com/gwi/service/TaskServiceImpl.scala

```diff
@@ -0,0 +1,57 @@
+package com.gwi.service
+
+import akka.NotUsed
+import akka.actor.{ActorSystem, Cancellable}
+import akka.http.scaladsl.model.Uri
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.gwi.execution.{Task, TaskExecutor}
+import com.gwi.api.{TaskDetail, TaskState}
+import com.gwi.repository.TaskRepository
+import com.gwi.storage.TaskStorage
+
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+class TaskServiceImpl(taskRepository: TaskRepository, taskStorage: TaskStorage, taskExecutor: TaskExecutor)(implicit
+    ec: ExecutionContext,
+    system: ActorSystem
+) extends TaskService {
+
+  private val taskStateRefreshInterval = 2.seconds
+
+  override def createTask(csvUri: Uri): Future[UUID] = {
+    val newTask = Task(UUID.randomUUID(), csvUri)
+    for {
+      _ <- taskRepository.insertTask(newTask)
+      _ <- taskExecutor.enqueueTask(newTask.id)
```

Comment:
I would recommend propagating errors in the effect channel otherwise they can be easily ignored as in this case.

Reply:
Thanks @pribor that's right. It has been already addressed.